src/frontend/src/lib/types/token.ts

```diff
@@ -41,7 +41,7 @@ export type OptionTokenStandard = TokenStandard | undefined | null;
 export type TokenToPin = Pick<Token, 'id'> & { network: Pick<Token['network'], 'id'> };
 
 interface TokenFinancialData {
-	balance?: BigNumber;
+	balance?: BigNumber | null;
```

Comment:
Just an idea that crossed my mind and not for this PR: maybe we can create a TypeScript util for those nullable types?

Reply:
similar to `mapAddress` you mean? generalize it?

```
export const mapAddress = <T extends Address>(
	$addressStore: StorageAddressData<T>
): OptionAddress<Address> => ($addressStore === null ? null : $addressStore?.data);
```

Reply:
No I meant a type, more like `RequiredExcept` or `ResultSuccessReduced`.

Reply:
Started working on it with:

https://github.com/dfinity/oisy-wallet/pull/2326